wicket-examples/src/main/java/org/apache/wicket/examples/websocket/progress/ProgressUpdater.java

```diff
@@ -0,0 +1,176 @@
+	{
+		// create an asynchronous task that will write the data to the client
+		ProgressUpdateTask progressUpdateTask = new ProgressUpdateTask(application, session);
+		scheduledExecutorService.schedule(progressUpdateTask, 1, TimeUnit.SECONDS);
+		return progressUpdateTask;
+	}
+
+	/**
+	 * Signal task was canceled.
+	 */
+	public static class TaskCanceled implements IWebSocketPushMessage
+	{
+	}
+
+	/**
+	 * A push message used to update progress.
+	 */
+	public static class ProgressUpdate implements IWebSocketPushMessage
+	{
+
+		private final int progress;
+
+		public ProgressUpdate(int progress)
+		{
+			this.progress = progress;
+		}
+
+		public int getProgress()
+		{
+			return progress;
+		}
+	}
+
+	/**
+	 * A task that sends data to the client by pushing it to the web socket connection
+	 */
+	public static class ProgressUpdateTask implements Runnable, Serializable
+	{
+		/**
+		 * The following fields are needed to be able to lookup the IWebSocketConnection from
+		 * IWebSocketConnectionRegistry
+		 */
+		private final String applicationName;
+		private final String sessionId;
+
+		private boolean canceled = false;
+		private boolean running = false;
+
+		private ProgressUpdateTask(Application application, String sessionId)
+		{
+			this.applicationName = application.getName();
+			this.sessionId = sessionId;
+		}
+
+		@Override
+		public void run()
+		{
+			running = true;
+			Application application = Application.get(applicationName);
+			WebSocketSettings webSocketSettings = WebSocketSettings.Holder.get(application);
+
+			int progress = 0;
+
+			while (progress <= 100)
+			{
+				try
+				{
+					WebSocketPushBroadcaster broadcaster =
+							new WebSocketPushBroadcaster(webSocketSettings.getConnectionRegistry());
+
+					if (canceled)
+					{
+						canceled = false;
+						running = false;
+						broadcaster.broadcastAllMatchingFilter(application, (sessionId, key) ->
+								ProgressUpdateTask.this.sessionId.equals(sessionId) && key instanceof PageIdKey
+										&& ITaskProgressListener.class.isAssignableFrom(getPageClass(key)),
+								new TaskCanceled());
+						return;
+					}
+					broadcaster.broadcastAllMatchingFilter(application, (sessionId, key) ->
+							ProgressUpdateTask.this.sessionId.equals(sessionId) && key instanceof PageIdKey &&
+									ITaskProgressListener.class.isAssignableFrom(getPageClass(key)),
+							new ProgressUpdate(progress));
+
+					// sleep for a while to simulate work
+					TimeUnit.SECONDS.sleep(1);
+					progress++;
+				}
+				catch (InterruptedException x)
+				{
+					Thread.currentThread().interrupt();
+					break;
+				}
+				catch (Exception e)
+				{
+					LOGGER.error("unexpected exception", e);
+					break;
+				}
+			}
+			running = false;
+		}
+
+		protected Class<?> getPageClass(IKey iKey) {
+			try {
+				return Thread.currentThread().getContextClassLoader().loadClass(iKey.getContext());
```

Comment:
At line 110 we get the Application. Let's use it to get the IClassResolver

Reply:
Done. Using class resolver now